frontend/tests/controllers/CourseCloneTest.php

```diff
@@ -134,4 +134,109 @@ public function testCreateCourseCloneWithTheSameAlias() {
             'start_time' => Time::get()
         ]));
     }
+
+    /**
+     * Create clone of a course with problems that have been changed their
+     * visibility mode from public to private
+     */
+    public function testCreateCourseCloneWithPrivateProblems() {
+        $homeworkCount = 2;
+        $testCount = 2;
+        $problemsPerAssignment = 2;
+        $studentCount = 2;
+        $problemAssignmentsMap = [];
+
+        // Create course with assignments
+        $courseData = CoursesFactory::createCourseWithNAssignmentsPerType([
+            'homework' => $homeworkCount,
+            'test' => $testCount
+        ]);
+
+        // Add problems to assignments
+        $adminLogin = self::login($courseData['admin']);
+        for ($i = 0; $i < $homeworkCount + $testCount; $i++) {
+            $assignmentAlias = $courseData['assignment_aliases'][$i];
+            $problemAssignmentsMap[$assignmentAlias] = [];
+
+            for ($j = 0; $j < $problemsPerAssignment; $j++) {
+                $problemData = ProblemsFactory::createProblem();
+                CourseController::apiAddProblem(new Request([
+                    'auth_token' => $adminLogin->auth_token,
+                    'course_alias' => $courseData['course_alias'],
+                    'assignment_alias' => $assignmentAlias,
+                    'problem_alias' => $problemData['request']['problem_alias'],
+                ]));
+                $problemAssignmentsMap[$assignmentAlias][] = $problemData;
+            }
+        }
+        foreach ($problemAssignmentsMap as $assignment => $problems) {
```

Comment:
nit: 

```suggestion
        foreach ($assignmentProblemsMap as $assignment => $problems) {
```

ese nombre tiene un poquitín más de consistencia porque hace el mapeo de un assignment a varios problemas.

Reply:
Listo